htdocs/bom/class/bom.class.php

```diff
@@ -94,7 +94,7 @@ class BOM extends CommonObject
 	 * @var array  Array with all fields and their property. Do not use it as a static var. It may be modified by constructor.
 	 */
 	public $fields = array(
-		'rowid' => array('type'=>'integer', 'label'=>'TechnicalID', 'enabled'=>1, 'visible'=>-1, 'position'=>1, 'notnull'=>1, 'index'=>1, 'comment'=>"Id",),
+		'rowid' => array('type'=>'integer', 'label'=>'TechnicalID', 'enabled'=>1, 'visible'=>0, 'position'=>1, 'notnull'=>1, 'index'=>1, 'comment'=>"Id",),
```

Comment:
In module builder, we use -2 for the value 'visible' for 'rowid'. Can you try with it instead of 0 ?
diff is that with -2, property should also be visible in list (with 0, it is never visible)